src/Avalonia.Base/Media/GeometryGroup.cs

```diff
@@ -10,23 +10,20 @@ namespace Avalonia.Media
     /// </summary>
     public class GeometryGroup : Geometry
     {
-        public static readonly DirectProperty<GeometryGroup, GeometryCollection> ChildrenProperty =
-            AvaloniaProperty.RegisterDirect<GeometryGroup, GeometryCollection> (
-                nameof(Children),
-                o => o.Children,
-                (o, v)=> o.Children = v);
+        public static readonly StyledProperty<GeometryCollection> ChildrenProperty = 
+            AvaloniaProperty.Register<GeometryGroup, GeometryCollection>(nameof(Children));
 
         public static readonly StyledProperty<FillRule> FillRuleProperty =
             AvaloniaProperty.Register<GeometryGroup, FillRule>(nameof(FillRule));
 
-        private GeometryCollection _children;
-
         public GeometryGroup()
         {
-            _children = new GeometryCollection
-            {
-                Parent = this
-            };
+            SetCurrentValue(ChildrenProperty, new() { Parent = this });
```

Comment:
In particular, I'm afraid this pattern of initializing collection properties isn't a good idea. Consider the following test:

```
        [Fact]
        public void Styled_Collection_Property_Problem()
        {
            var target = new GeometryGroup();
            target.Children.Add(new RectangleGeometry());

            Assert.Single(target.Children);

            // Simulate a style with a trigger becoming active on the target.
            var d = target.SetValue(GeometryGroup.ChildrenProperty, new GeometryCollection(), Data.BindingPriority.StyleTrigger);
            Assert.Empty(target.Children);

            // Simulate the style becoming inactive
            d.Dispose();

            // Original value is lost
            Assert.Single(target.Children);
        }
```

Reply:
I guess that's why all collections are read-only in WPF.

I reverted all those changes. I just overlooked that folder when reverting the animatable types.